src/github.com/getlantern/peerscanner/host.go

```diff
@@ -346,10 +350,10 @@ func (h *host) deregisterHost() {
 }
 
 func (h *host) doDeregisterHost() error {
+	err := cfutil.DestroyRecord(h.record)
 	h.record = nil
 	h.isProxying = false
 
-	err := cfutil.DestroyRecord(h.record)
```

Comment:
Opportunistic fix.  Previously, destroying the h.record never worked because we had already cleared it.